qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/MixedClusterTestNodes.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.backwards;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.test.rest.ObjectPath;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+final class MixedClusterTestNodes {
+    private final Map<String, MixedClusterTestNode> nodesById;
+    private final String bwcNodesVersion;
+
+    private MixedClusterTestNodes(String bwcNodesVersion, Map<String, MixedClusterTestNode> nodesById) {
+        this.bwcNodesVersion = bwcNodesVersion;
+        this.nodesById = nodesById;
+    }
+
+    public List<MixedClusterTestNode> getNewNodes() {
+        return nodesById.values().stream().filter(n -> n.version().equals(bwcNodesVersion) == false).collect(Collectors.toList());
+    }
+
+    public List<MixedClusterTestNode> getBWCNodes() {
+        return nodesById.values().stream().filter(n -> n.version().equals(bwcNodesVersion)).collect(Collectors.toList());
+    }
+
+    public MixedClusterTestNode getSafe(String id) {
+        MixedClusterTestNode node = nodesById.get(id);
+        if (node == null) {
+            throw new IllegalArgumentException("node with id [" + id + "] not found");
+        }
+        return node;
+    }
+
+    static MixedClusterTestNodes buildNodes(RestClient client, String bwcNodesVersion) throws IOException {
+        Response response = client.performRequest(new Request("GET", "_nodes"));
+        ObjectPath objectPath = ObjectPath.createFromResponse(response);
+        Map<String, Object> nodesAsMap = objectPath.evaluate("nodes");
+
+        Map<String, MixedClusterTestNode> nodesById = new HashMap<>();
```

Comment:
Could use streams and `Collectors.toUnmodifiableMap` here

Reply:
That was my first implementation, but since `objectPath.evaluate` throws code in the `map` lambda would become messy (try-catch-rethrow) and I preferred to leave it as-is.

Reply:
Ah, hadn't spotted the exception

Reply:
Neither did I, I was "disappointed" when I have to go back to a for loop. Oh well.